user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -387,14 +391,22 @@ class RapidsJarTool(RapidsTool):
     """
 
     def _process_jar_arg(self):
+        jar_path = ''
         tools_jar_url = self.wrapper_options.get('toolsJar')
-        if tools_jar_url is None:
-            tools_jar_url = self.ctxt.get_rapids_jar_url()
-        # download the jar
-        jar_path = self.ctxt.platform.storage.download_resource(tools_jar_url,
-                                                                self.ctxt.get_local_work_dir(),
-                                                                fail_ok=False,
-                                                                create_dir=True)
+        try:
+            tools_jar_url = None
+            if tools_jar_url is None:
+                tools_jar_url = self.ctxt.get_rapids_jar_url()
+            # download the jar
+            self.logger.info('Downloading the tools jars %s', tools_jar_url)
+            jar_path = self.ctxt.platform.storage.download_resource(tools_jar_url,
```

Comment:
Should we add the try-except block in `download_resource()` for consistent error handling when downloading resources in the codebase? 

We could use the `fail_ok` flag to control the except block.

Reply:
yeah I debated that. It would probably be good to have at least a log message there.  can you put a comment in the issue 1286 and we can do it as a followup. I wanted to understand more how all the fail_ok stuff was really used before making that change

Reply:
Sure Tom. Added it as a follow up comment https://github.com/NVIDIA/spark-rapids-tools/issues/1286#issuecomment-2293806888